src/main/coordinator.c

```diff
@@ -135,6 +135,22 @@ coordinator_worker(void* udata)
 		stage_t* stage = &cdata->stages.stages[stage_idx];
 		fprint_stage(stdout, &cdata->stages, stage_idx);
 
+		if (stage->workload.type == WORKLOAD_TYPE_I) {
+			uint64_t nkeys = stage->key_end - stage->key_start;
+			if (stage->async) {
+				if (nkeys % stage->batch_write_size != 0) {
+					blog_warn("--keys is not divisible by --batch-write-size so more than "
+								"--keys records will be written\n");
+				}
+			}
+			else {
+				if (stage->batch_write_size * n_threads > nkeys) {
+					blog_warn("--batch-write-size * --threads is greater than --keys so "
+								"more than --keys records will be written\n");
```

Comment:
Each thread writes a minimum of --batch-write-size records. I chose to go this route because it simplifies batch size handling. When I come back to this PR I'll take another look at splitting batch-size according to --keys.